core/storage/merkle/node.hpp

```diff
@@ -6,23 +6,15 @@
 #ifndef KAGOME_NODE_HPP
 #define KAGOME_NODE_HPP
 
-#include "common/buffer.hpp"
-
 namespace kagome::storage::merkle {
 
-  using common::Buffer;
+  struct Node {
+    enum class Type { Leaf, Branch };
 
-  enum class NodeType : uint8_t {
-    LEAF = 1,
-    EXTENSION = 128,
-    BRANCH_NO_VALUE = 254,
-    BRANCH_WITH_VALUE = 255
-  };
+    virtual ~Node() = default;
 
-  class Node {
-   public:
     // returns type of a node
-    virtual NodeType getType() const = 0;
+    virtual Type getType() const = 0;
```

Comment:
Is this method needed in a struct? Are you going to make private `Type` variable in successors?

Reply:
this method is a way to get type of `node` without RTTI, so yes, it is needed